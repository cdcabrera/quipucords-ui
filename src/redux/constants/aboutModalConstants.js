const ABOUT_MODAL_SHOW = 'ABOUT_MODAL_SHOW';
const ABOUT_MODAL_HIDE = 'ABOUT_MODAL_HIDE';

const aboutModalTypes = {
  ABOUT_MODAL_SHOW,
  ABOUT_MODAL_HIDE
};

export { aboutModalTypes as default, aboutModalTypes, ABOUT_MODAL_SHOW, ABOUT_MODAL_HIDE };
