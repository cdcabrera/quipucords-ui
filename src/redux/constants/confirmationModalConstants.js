const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

const confirmationModalTypes = {
  CONFIRMATION_MODAL_SHOW,
  CONFIRMATION_MODAL_HIDE
};

export { confirmationModalTypes as default, confirmationModalTypes, CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
