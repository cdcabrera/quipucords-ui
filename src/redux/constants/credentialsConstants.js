const ADD_CREDENTIAL = 'ADD_CREDENTIAL';
const DELETE_CREDENTIAL = 'DELETE_CREDENTIAL';
const GET_CREDENTIALS = 'GET_CREDENTIALS';
const UPDATE_CREDENTIAL = 'UPDATE_CREDENTIAL';
const CREATE_CREDENTIAL_SHOW = 'CREATE_CREDENTIAL_SHOW';
const EDIT_CREDENTIAL_SHOW = 'EDIT_CREDENTIAL_SHOW';
const UPDATE_CREDENTIAL_HIDE = 'UPDATE_CREDENTIAL_HIDE';

const SELECT_CREDENTIAL = 'SELECT_CREDENTIAL';
const DESELECT_CREDENTIAL = 'DESELECT_CREDENTIAL';
const EXPANDED_CREDENTIAL = 'EXPANDED_CREDENTIAL';
const NOT_EXPANDED_CREDENTIAL = 'NOT_EXPANDED_CREDENTIAL';
const UPDATE_CREDENTIALS = 'UPDATE_CREDENTIALS';

const credentialsTypes = {
  ADD_CREDENTIAL,
  UPDATE_CREDENTIAL,
  DELETE_CREDENTIAL,
  GET_CREDENTIALS,
  CREATE_CREDENTIAL_SHOW,
  EDIT_CREDENTIAL_SHOW,
  UPDATE_CREDENTIAL_HIDE,
  SELECT_CREDENTIAL,
  DESELECT_CREDENTIAL,
  EXPANDED_CREDENTIAL,
  NOT_EXPANDED_CREDENTIAL,
  UPDATE_CREDENTIALS
};

export {
  credentialsTypes as default,
  credentialsTypes,
  ADD_CREDENTIAL,
  UPDATE_CREDENTIAL,
  DELETE_CREDENTIAL,
  GET_CREDENTIALS,
  CREATE_CREDENTIAL_SHOW,
  EDIT_CREDENTIAL_SHOW,
  UPDATE_CREDENTIAL_HIDE,
  SELECT_CREDENTIAL,
  DESELECT_CREDENTIAL,
  EXPANDED_CREDENTIAL,
  NOT_EXPANDED_CREDENTIAL,
  UPDATE_CREDENTIALS
};
