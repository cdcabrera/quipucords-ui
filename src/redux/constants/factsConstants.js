const ADD_FACTS = 'ADD_FACTS';

const factsTypes = {
  ADD_FACTS
};

export { factsTypes as default, factsTypes, ADD_FACTS };
