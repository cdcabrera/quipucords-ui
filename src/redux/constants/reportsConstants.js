const GET_REPORTS_DOWNLOAD = 'GET_REPORTS_DOWNLOAD';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORTS_DOWNLOAD, GET_MERGE_REPORT };
