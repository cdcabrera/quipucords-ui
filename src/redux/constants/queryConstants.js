import { apiTypes } from '../../constants/apiConstants';

/**
 * Inventory query types associated with only HOSTS' and INSTANCES' queries.
 */
const SET_QUERY_CREDENTIALS_TYPES = {
  [apiTypes.API_.DIRECTION]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.DISPLAY_NAME]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DISPLAY_NAME}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};
