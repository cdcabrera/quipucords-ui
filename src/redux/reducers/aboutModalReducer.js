import { aboutModalTypes } from '../constants';
import { reduxHelpers } from '../common/reduxHelpers';

const initialState = {
  show: false
};

const aboutModalReducer = (state = initialState, action) => {
  switch (action.type) {
    case aboutModalTypes.ABOUT_MODAL_SHOW:
      return reduxHelpers.setStateProp(
        null,
        {
          show: true
        },
        {
          state,
          reset: false
        }
      );

    case aboutModalTypes.ABOUT_MODAL_HIDE:
      return reduxHelpers.setStateProp(
        null,
        {
          show: false
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

aboutModalReducer.initialState = initialState;

export { aboutModalReducer as default, initialState, aboutModalReducer };
