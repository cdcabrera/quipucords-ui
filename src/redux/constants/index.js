import { aboutModalTypes } from './aboutModalConstants';
import { confirmationModalTypes } from './confirmationModalConstants';
import { credentialsTypes } from './credentialsConstants';
import { factsTypes } from './factsConstants';
import { reportsTypes } from './reportsConstants';
import { scansTypes } from './scansConstants';
import { sourcesTypes } from './sourcesConstants';
import { statusTypes } from './statusConstants';
import { toastNotificationTypes } from './toasNotificationConstants';
import { userTypes } from './userConstants';
import { viewTypes } from './viewConstants';
import { viewPaginationTypes } from './viewPaginationConstants';
import { viewToolbarTypes } from './viewToolbarConstants';

const reduxTypes = {
  aboutModal: aboutModalTypes,
  confirmationModal: confirmationModalTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  status: statusTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes,
  view: viewTypes,
  viewPagination: viewPaginationTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  aboutModalTypes,
  confirmationModalTypes,
  credentialsTypes,
  factsTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  statusTypes,
  toastNotificationTypes,
  userTypes,
  viewTypes,
  viewPaginationTypes,
  viewToolbarTypes
};
