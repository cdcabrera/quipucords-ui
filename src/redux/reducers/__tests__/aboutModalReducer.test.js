import aboutModalReducer from '../aboutModalReducer';
import { aboutModalTypes as types } from '../../constants';

describe('AddSourceWizardReducer', () => {
  it('should return the initial state', () => {
    expect(aboutModalReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [types.ABOUT_MODAL_HIDE, types.ABOUT_MODAL_SHOW];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value,
        source: {}
      };

      const resultState = aboutModalReducer(undefined, dispatched);

      expect({
        type: value,
        result: resultState
      }).toMatchSnapshot(`defined type ${value}`);
    });
  });
});
