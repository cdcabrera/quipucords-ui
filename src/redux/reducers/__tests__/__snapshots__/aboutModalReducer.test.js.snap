// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSourceWizardReducer should handle specific defined types: defined type ABOUT_MODAL_HIDE 1`] = `
Object {
  "result": Object {
    "show": false,
  },
  "type": "ABOUT_MODAL_HIDE",
}
`;

exports[`AddSourceWizardReducer should handle specific defined types: defined type ABOUT_MODAL_SHOW 1`] = `
Object {
  "result": Object {
    "show": true,
  },
  "type": "ABOUT_MODAL_SHOW",
}
`;
